### PR TITLE
MIC-686: UaR: Enter review details

### PR DESCRIPTION
**Pull request by JackReeveMoJ**
image

### CODE REVIEW

File: integration_tests/mockApis/csipApi.ts [hunk 1]
```diff
@@ -764,6 +764,10 @@ const stubPostIdentifiedNeedFail = () => {
   })
 }
 
+const stubPatchReviewSuccess = () => {
+  return createBasicHttpStub('PATCH', '/csip-api/csip-records/plan/reviews/[a-zA-Z0-9-]+-[a-zA-Z0-9-]+', 200)
+}
```
Comment (bill-moj) [inline integration_tests/mockApis/csipApi.ts:769]: `[a-zA-Z0-9-]+-[a-zA-Z0-9-]+` do we need to repeat it twice?

File: server/routes/journeys/record-review/details/view.njk [hunk 2]
```diff
@@ -12,35 +12,28 @@
...
+                <li>a summary of progress on each identified need</li>
+                <li>information on any set-backs or changes in risk</li>
+            </ul>'
+        }) }}
         {{ govukCharacterCount({
-            maxlength: maxLengthChars,
+            maxlength: 4000,
             rows: 12,
-            threshold: threshold,
+            threshold: 75,
```
Comment (bill-moj) [inline server/routes/journeys/record-review/details/view.njk:36]: you might remove maxLength and threshold from `record-review/details/controller`

File: server/routes/journeys/update-review/details/controller.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      recordUuid: req.journeyData.csipRecord!.recordUuid,
+    })
+  }
+
+  checkSubmitToAPI = async (req: Request<unknown, unknown, SchemaType>, _: Response, next: NextFunction) =>
+    this.submitChanges({
+      req,
+      next,
+      changes: {
+        summary: req.body.summary ?? '',
```
Comment (bill-moj) [inline server/routes/journeys/update-review/details/controller.ts:22]: `req.body.summary` should be non-nullable, we don't need `??` on it?

File: server/routes/journeys/update-review/details/tests.cy.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+  it('should try out all cases', () => {
+    navigateToTestPage()
+    cy.url().should('to.match', /\/details$/)
+    checkAxeAccessibility()
+
+    validatePageContents()
+    validateErrorMessage()
+    proceedToNextScreen()
+  })
```
Comment (bill-moj) [inline server/routes/journeys/update-review/details/tests.cy.ts:29]: might worth adding a stubPatchReviewFail to test for API failure scenario too

File: integration_tests/mockApis/csipApi.ts (hunk 1 above)
Comment (JackReeveMoJ) [reply integration_tests/mockApis/csipApi.ts:769]: so that it doesn't capture the `/attendees` endpoint as well

**Review by bill-moj — APPROVED** [review]

**Pull request merged**